systemvm/agent/noVNC/app/ui.js

```diff
@@ -1206,38 +1212,14 @@ const UI = {
  * ------v------*/
 
     toggleFullscreen() {
-        if (document.fullscreenElement || // alternative standard method
-            document.mozFullScreenElement || // currently working methods
-            document.webkitFullscreenElement ||
-            document.msFullscreenElement) {
-            if (document.exitFullscreen) {
-                document.exitFullscreen();
-            } else if (document.mozCancelFullScreen) {
-                document.mozCancelFullScreen();
-            } else if (document.webkitExitFullscreen) {
-                document.webkitExitFullscreen();
-            } else if (document.msExitFullscreen) {
-                document.msExitFullscreen();
-            }
-        } else {
-            if (document.documentElement.requestFullscreen) {
-                document.documentElement.requestFullscreen();
-            } else if (document.documentElement.mozRequestFullScreen) {
-                document.documentElement.mozRequestFullScreen();
-            } else if (document.documentElement.webkitRequestFullscreen) {
-                document.documentElement.webkitRequestFullscreen(Element.ALLOW_KEYBOARD_INPUT);
-            } else if (document.body.msRequestFullscreen) {
-                document.body.msRequestFullscreen();
-            }
-        }
-        UI.updateFullscreenButton();
+        this.fullScreen = !this.fullScreen
+        UI.rfb.scaleViewport = this.fullScreen
```

Comment:
@davidjumani 
one of our customers has issue accessing the novnc console, because scaleViewport is undefined in line 1216
`
        UI.rfb.scaleViewport = this.fullScreen
`
But I cannot find this change in novnc repository.
https://github.com/novnc/noVNC/blob/master/app/ui.js#L1209

Reply:
@weizhouapache I've changed a bit of the novnc UI to allow fullscreen and pasting into the console.
fullscreen is defined at line 44 and scaleViewport is defined in rfb.js

Reply:
> @weizhouapache I've changed a bit of the novnc UI to allow fullscreen and pasting into the console.
> fullscreen is defined at line 44 and scaleViewport is defined in rfb.js

@davidjumani 
we do not have any problem in our testing. but that customer cannot access the console. he tried with different browsers. 
it is strange. it seem UI.rfb is not initialized.

Reply:
@weizhouapache That could be a bigger issue since rfb.js is used to communicate with the host. Did destroying and recreating the console proxy help ? Also could you check whether the customer has tweaked the novnc code in the cpvm ? Or any issue with the systemvm.iso itself ?

Reply:
> @weizhouapache That could be a bigger issue since rfb.js is used to communicate with the host. Did destroying and recreating the console proxy help ? Also could you check whether the customer has tweaked the novnc code in the cpvm ? Or any issue with the systemvm.iso itself ?

@davidjumani ignore the issue above. It happened when we enlarge the screen when websocket connection is not working. it is a bug but not critical. we are still looking for the root cause of the issue. customer uses German keyboard which is different from us.

Reply:
@weizhouapache Fixed in https://github.com/apache/cloudstack/pull/4783
It prevents the fullscreen button from being shown when not connected to the VM